nix/overlays.nix

```diff
@@ -2,25 +2,32 @@ self: super: {
   python27 = super.python27.override {
     packageOverrides = python-self: python-super: {
       # eliot is not part of nixpkgs at all at this time.
-      eliot = python-self.callPackage ./eliot.nix { };
+      eliot = python-self.pythonPackages.callPackage ./eliot.nix { };
 
       # NixOS autobahn package has trollius as a dependency, although
       # it is optional. Trollius is unmaintained and fails on CI.
-      autobahn = python-super.callPackage ./autobahn.nix { };
+      autobahn = python-super.pythonPackages.callPackage ./autobahn.nix { };
 
       # Porting to Python 3 is greatly aided by the future package.  A
       # slightly newer version than appears in nixos 19.09 is helpful.
-      future = python-super.callPackage ./future.nix { };
+      future = python-super.pythonPackages.callPackage ./future.nix { };
 
       # Need version of pyutil that supports Python 3. The version in 19.09
       # is too old.
-      pyutil = python-super.callPackage ./pyutil.nix { };
+      pyutil = python-super.pythonPackages.callPackage ./pyutil.nix { };
 
       # Need a newer version of Twisted, too.
-      twisted = python-super.callPackage ./twisted.nix { };
+      twisted = python-super.pythonPackages.callPackage ./twisted.nix { };
 
       # collections-extended is not part of nixpkgs at this time.
-      collections-extended = python-super.callPackage ./collections-extended.nix { };
+      collections-extended = python-super.pythonPackages.callPackage ./collections-extended.nix { };
+    };
+  };
+
+  python39 = super.python39.override {
```

Comment:
For `collections-extended` v1.0.3 is Py2+Py3, but there is newer release that is Python 3 only. I hardcoded 1.0.3 for Python 2. but perhaps Python 3 version can use newer version? And the same for other dependencies more broadly.

Reply:
Probably so.  Fortunately all of the other dependencies are already in nixpkgs for Python 3 now!  I think a sensible plan is to get `collections-extended` in, too, and then declare we require at least that version of nixpkgs ... then we'll be down to zero overrides for Python 3 (at least until we adopt another unpackaged dependency :smile: ).

Unless there are known issues (and the test suite suggests no) w/ this version on Py 3 I'd rather just re-use the version you already packaged for Python 2 rather than carry two package versions in Tahoe.

Reply:
OK.